nodeprobe/nodeprobe.go

```diff
@@ -136,3 +136,23 @@ func (p *Prober) AddNode(name string, node Node) {
 
 	p.nodes[name] = node
 }
+
+// TODO: string constants are error-prone.
+// Add a method to clients that returns their name or solve this in another way
+func (p *Prober) CheckBeaconNodeHealth(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, p.interval)
+	defer cancel()
+	return p.nodes["consensus client"].Healthy(ctx)
+}
+
+func (p *Prober) CheckExecutionNodeHealth(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, p.interval)
+	defer cancel()
+	return p.nodes["execution client"].Healthy(ctx)
+}
+
+func (p *Prober) CheckEventSyncerHealth(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, p.interval)
+	defer cancel()
+	return p.nodes["event syncer"].Healthy(ctx)
+}
```

Comment:
all reset of the places accessing the nodes are using mutex, shouldn't we protect here as well?

Reply:
added mutex